### PR TITLE
Updated go-marathon to fully support the Queue API

### PR DESCRIPTION
**Pull request by kiryojo**
Marathon Docs:
http://mesosphere.github.io/marathon/api-console/index.html

Most of the work was just adding the structs that represent the fields returned by the Queue API.  Some fields were also modified, such as "dependencies", because I found evidence (in both the documentation and from API calls) that those fields are omitted when empty.

### CODE REVIEW

File: queue.go [hunk 1]
```diff
@@ -18,24 +18,77 @@ package marathon
...
+	"time"
 )
 
-// Queue is the definition of marathon queue
 type Queue struct {
 	Items []Item `json:"queue"`
 }
 
-// Item is the definition of element in the queue
+// Queue is the definition of marathon queue
```
Comment (jeffizhungry) [inline queue.go]: Looks like you moved some comments

File: error.go [hunk 2]
```diff
@@ -58,7 +58,7 @@ func (e *InvalidEndpointError) Error() string {
 
 // newInvalidEndpointError creates a new error
 func newInvalidEndpointError(message string, args ...interface{}) error {
-	return &InvalidEndpointError{message: fmt.Sprintf(message, args)}
+	return &InvalidEndpointError{message: fmt.Sprintf(message, args...)}
```
Comment (jeffizhungry) [inline error.go]: LOL

Comment (kiryojo) [reply error.go]: Thanks Golang 1.12!

File: queue.go (hunk 1 above)
Comment (kiryojo) [reply queue.go]: Oh right I should adjust some of the comments.

**Comment by kiryojo** [comment]
> Do you plan to submit this back upstream too?

Good call, I'll look into it

**Pull request merged**